source/extensions/stat_sinks/metrics_service/grpc_metrics_service_impl.h

```diff
@@ -23,40 +23,43 @@ namespace MetricsService {
 /**
  * Interface for metrics streamer.
  */
-class GrpcMetricsStreamer
-    : public Grpc::AsyncStreamCallbacks<envoy::service::metrics::v3::StreamMetricsResponse> {
+template <class ResponseProto>
```

Comment:
I wonder if I should template on RequestProto as well and move some of the member variables in the impl (stream_, client_) to the base class.

Reply:
I would probably template on both and pass the service name as a ctor param, atm it's hard coded to 

```
      service_method_(
          Grpc::VersionedMethods("envoy.service.metrics.v3.MetricsService.StreamMetrics",
                                 "envoy.service.metrics.v2.MetricsService.StreamMetrics")
```